src/ServerScriptService/Services/WeldService.luau

```diff
@@ -0,0 +1,63 @@
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local Graph = require(ReplicatedStorage.Common.Graph)
+local Knit = require(ReplicatedStorage.Packages.Knit)
+
+local WeldService = Knit.CreateService {
+    Name = "WeldService";
+    Client = {};
+}
+
+function WeldService:WeldModel(model)
+    local primaryPart = assert(model.PrimaryPart, "Model has no primary part!")
+
+    for _, v in ipairs(model:GetChildren()) do
+        local weld = Instance.new("Weld")
+        weld.Part0 = primaryPart
+        weld.Part1 = v
+        weld.Parent = model
+    end
+end
+
+function WeldService:RemoveModel(model)
+    for otherModel, weld in self._weldGraph:GetNodeConnections(model) do
+        weld:Destroy()
+        self._weldGraph:Disconnect(model, otherModel)
+    end
+
+    self._connections[model]:Disconnect()
+    self._connections[model] = nil
+end
+
+function WeldService:WeldModels(model1, model2)
+    assert(model1.PrimaryPart, "Model has no primary part!")
+    assert(model2.PrimaryPart, "Model has no primary part!")
+
+    if self._weldGraph:GetNodeConnections(model1)[model2] then
+        return
+    end
+
+    if not self._connections[model1] then
+        model1.Destroying:Connect(function() self:RemoveModel(model1) end)
+    end
+
+    if not self._connections[model2] then
+        model2.Destroying:Connect(function() self:RemoveModel(model2) end)
+    end
```

Comment:
We need to actually store the connection in the self._connections table. This seems to be just a slip up, and not intended.